src/v/cluster/data_migration_service_handler.cc

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "cluster/data_migration_service_handler.h"
+
+#include "cluster/data_migration_frontend.h"
```

Comment:
this commit won't seem to build, as this header is only introduced in a subsequent commit

Reply:
>this commit won't seem to build, as this header is only introduced in a subsequent commit

nice catch!